lib/Runtime/PlatformAgnostic/RuntimePlatformAgnosticPch.h

```diff
@@ -15,6 +17,42 @@
 #pragma warning(pop)
 #endif
 
+// Minimal definitions to use AssertMsg in the PAIL
+#ifndef _WIN32
+#define DbgRaiseAssertionFailure() __builtin_trap()
+#define __analysis_assume(x)
+#define __in
+#define __inout
+
+#ifndef TRUE
```

Comment:
This is supposed to be the layer that replaces the PAL long term (hence called Runtime/PlatformAgnostic). It would be odd for it to take a dependency on the PAL.